### ISSUE TITLE
Checked ContextualMenuItems don't have their position announced

### ISSUE THREAD

**Issue by Debml** [microsoft/fluentui#18487]
### Environment Information

- **Package version(s)**: @fluentui/react 8.17.1
- **Browser and OS versions**: Edge 91.0.864.41 (64-bit), Win 10 Enterprise Version 21H1 OS build 19043.985

### Describe the issue:

When rendering contextual menu items that can be checked (setting checked or canCheck props), screen readers don't announce the position of the menu items relative to other siblings.

When the checked props are not set, screen readers correctly announce something like "Menu item 1 of 4". 

Inspecting the HTML shows that the "aria-posinset" and "aria-setsize" attributes are set correctly, but they aren't announced. Items also seem to be under a "Menu" role item. Seems that the only difference is the role of the button, which is "menuitemcheckbox" in these cases, and a regular "menuitem" when it does work.

### Please provide a reproduction of the issue in a codepen:

https://codepen.io/debml700/pen/OJpwxEw

#### Actual behavior:

Narrator announces the role, name and checked state of the menuitemcheckbox, but it doesn't announce the position of the menuitemcheckbox relative to its siblings.

#### Expected behavior:

Narrator announces the role, name, checked state and position of the menuitemcheckbox relative to its siblings.

### Documentation describing expected behavior

https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html

**Comment by gouttierre** [comment]
@Debml - Thanks for reaching out and filing this issue with repro details.

@ling1726 -  Would you be able to confirm if this is a regression, or if this behavior is an issue?
cc @smhigley

**Comment by ling1726** [comment]
The issue here is that `FocusZone` needs a `role="presentation"` attribute. This is the current DOM output. The output is a far cry from [the recommended WAI menu](https://www.w3.org/TR/wai-aria-practices-1.1/examples/menubar/menubar-2/menubar-2.html). The proposed fix will resolve this issue, but it is recommended to clean up the amount of wrapping elements here in the future

```html
<div role="menu>
  <!-- Need to add role="presentation" below -->
  <div class="focuszone">
    <ul role="presentation">
      <li role="presentation">
        <div role="group">
          <ul role="presentation>
            <li role="presentation">
              <button role="menuitemcheckbox />
            </li>
          </ul>
        </div>
      </li>
    </ul>  
</div>
</div>
```

**Comment by ling1726** [comment]
@JustSlone please review the proposed fix for the backlog of the v8 team
